Support space in long and short options with spaces in deps

cryvate opened the pull request:
Add the moment, one cannot use options in deps like one does on the command line for pip:

`deps = -r foo`

gives an error, due to pip trying interpret `-r foo` as a requirement. This goes for all options that have an argument. The current situation is that on either has no space (short option) or an equal sign (long option). This is inconsistent with pip install itself, 

Previously, these would have given an error when trying to run `pip install`, now it will pass. This fixes #658.

For background, the arguments are processed again in the [_install method of class Virtualenv](https://github.com/tox-dev/tox/blob/f1a933b59c2682f40054c0cabe8c1f42a3635168/tox/venv.py#L311) and then in [method run_install_command it is given popen](https://github.com/tox-dev/tox/blob/f1a933b59c2682f40054c0cabe8c1f42a3635168/tox/venv.py#L307).

RonnyPfannschmidt commented:
i believe this one needs some unit and integration tests, at least a integration test tho

cryvate commented:
@RonnyPfannschmidt I agree, I will have a look later.

I did not run the tests (I was on the train with flaky connection, so running tox/pip was not easy) so I need to fix my use of .format to be py2.6 compatible.

It isn't nice to have to explicitly list all pip options with arguments. [Another road to travel would be to use shlex.split (available in 2.6 as well).](https://docs.python.org/3/library/shlex.html#shlex.split) This means the wheel is not being re-invented, on the other hand, it might do things that breaks backwards compatibility.

asottile commented:
It would be nice to use `shlex.split` but it turns out it's horribly broken in python2.6:

```
Python 2.6.9 (default, Jul 29 2017, 12:47:14) 
[GCC 5.4.0 20160609] on linux4
Type "help", "copyright", "credits" or "license" for more information.
>>> import shlex
>>> shlex.split(u'hello world')
['h\x00\x00\x00e\x00\x00\x00l\x00\x00\x00l\x00\x00\x00o\x00\x00\x00', '\x00\x00\x00w\x00\x00\x00o\x00\x00\x00r\x00\x00\x00l\x00\x00\x00d\x00\x00\x00']
```

gaborbernat commented on tox/config.py:
```diff
@@ -124,6 +136,20 @@ def postprocess(self, testenv_config, value):
+                # in case of a long option, we add an equal sign
+                for option in PIP_INSTALL_LONG_OPTIONS_ARGUMENT:
+                    if name.startswith(option + ' '):
+                        name = option + '=' + name[len(option):].strip()
```
I think we should to use string format notation for this

cryvate replied:
Good idea. Changed.

cryvate commented:
I think the travis build failed due to unrelated reasons? [See here.](https://travis-ci.org/tox-dev/tox/jobs/300680486)

gaborbernat commented:
rebase on master and it should be good now 👍

vlcinsky commented:
Really looking forward to this feature for quite some time as it now shall be simpler to jump to requirement file by using "gf" (go file) in my vim editor.

Thanks guys (and sorry for spoiling the issue).
